content/en/docs/making-requests.md

```diff
@@ -426,6 +426,195 @@ functional arguments to `NextPage`. For more information about overriding client
+    log.Printf("error: %v", err)
+    return 
+}
+fmt.Println("Dynamodb table is now ready for write operations")
+
+```
+
+Since the waiter constructor takes in a client interface, you can 
+provide a custom client that will be used by the waiter to poll the resource.
+This design also makes it easier to mock the waiter for testing purposes,
+
+#### Overriding waiter configuration
+
+By default, SDK uses the minimum delay and maximum delay value configured with 
+optimal values defined by AWS services for different APIs. You can override these
+configurations on the entire waiter or per request, via waiter-specific 
+functional options present for each waiter. 
+
+Here's how you can configure waiter-level override configuration
+
+```go
+import "context"
+import "fmt"
+import "log"
+import "time"
+import "github.com/aws/aws-sdk-go-v2/aws"
+import "github.com/aws/aws-sdk-go-v2/config"
+import "github.com/aws/aws-sdk-go-v2/service/dynamodb"
+
+// ...
+
+cfg, err := config.LoadDefaultConfig(context.TODO())
+if err != nil {
+    log.Printf("error: %v", err)
+    return
+}
+
+client := dynamodb.NewFromConfig(cfg)
+
+// we create a waiter instance by directly passing in a client
+// that satisfies the waiters client Interface. 
+waiter :=  dynamodb.NewTableExistsWaiter(client, func (o *dynamodb.TableExistsWaiterOptions) {
+	
+	// override minimum delay to 10 seconds
+	o.MinDelay = 10 * time.Second
+	
+	// override maximum default delay to 300 seconds
+	o.MaxDelay = 300 * time.Second
+})
+```
+
+The `Wait` function on each waiter also takes in waiter-specific functional options.  
+Similar to the above example, you can override waiter configuration per `Wait` request. 
+
+```go
+// params is the input to api operation used by the waiter
+params := &dynamodb.DescribeTableInput {
+	TableName: aws.String("test-table")
+}
+
+// maxWaitTime is the maximum wait time, the waiter will wait for 
+// the resource status.
+maxWaitTime := 5 * time.Minutes
+
+// Wait will poll until it gets the resource status, or max wait time 
+// expires.
+err := waiter.Wait(context.TODO(), params, maxWaitTime, func (o *dynamodb.TableExistsWaiterOptions) {
+
+    // override minimum delay to 5 seconds
+    o.MinDelay = 5 * time.Second
+
+    // override maximum default delay to 120 seconds
+    o.MaxDelay = 120 * time.Second
+})
+if err == nil {
+    log.Printf("error: %v", err)
+    return 
+}
+fmt.Println("Dynamodb table is now ready for write operations")
+
+```
+
+#### Advanced waiter configuration overrides
+
+In the new design of waiters, SDK provides advanced capabilities such as overriding how SDK determines, 
+if a waiter should retry based on the provided input or obtained api response. This functionality is 
+helpful for users who want to additionally customize and control the waiter behavior. The waiter-specific 
+options also provides `APIOptions` to [customize operation middlewares](https://aws.github.io/aws-sdk-go-v2/docs/middleware/#writing-a-custom-middleware).
```

Comment:
Since this is a configuration guide I wouldn't add language like "In the new design".

Maybe simplify it to something like: "You can override the waiter default behavior by providing a custom retryable function".